crates/spark/src/ssp/graphql/client.rs

```diff
@@ -105,14 +114,10 @@ impl GraphQLClient {
     where
         T: Serialize + Clone + Into<Q::Variables>,
     {
-        if needs_auth && !self.auth_provider.is_authorized().await? {
-            self.authenticate().await?;
-            tracing::debug!("Authenticated successfully with ssp");
-        }
```

Comment:
Looking at the calls to `post_query()` it seems all calls pass `needs_auth: true`. So I would say we could remove it. Otherwise this change is getting the session even if we might not need it